### PR TITLE
EDM-542: Fix enrollment request approver showing 'unknown' with JWT auth

### PR DESCRIPTION
**Pull request by galel12**
##  Problem

When approving enrollment requests via CLI in environments using JWT/OIDC authentication (e.g., OpenShift), the `APPROVER` field shows "unknown" instead of the actual authenticated username.

**Root Cause:** The `JWTAuth.GetIdentity()` method was not implemented, returning an empty `Identity` struct, causing the enrollment service to fall back to "unknown" as the approver.

## Solution

Implemented comprehensive JWT token parsing in `JWTAuth.GetIdentity()` to extract user identity from JWT claims:

### Key Changes:

1. **JWT Token Parsing** (`jwtverifer.go`):
   - Parse JWT tokens using `jwt.Parse()`
   - Extract claims as map for processing
   - Return structured `common.Identity` with username, UID, and groups

2. **JWT Claim Extraction Helpers** (`common.go`):
   - `ExtractUsernameFromJWTClaims()`: Extracts username from standard JWT claims
   - `ExtractUIDFromJWTClaims()`: Extracts user ID from claims  
   - `ExtractGroupsFromJWTClaims()`: Extracts user groups/roles
   - **Claim Priority:** `preferred_username` → `email` → `name` → `sub`

3. **Comprehensive Unit Tests** (`jwt_test.go`):
   - 276 lines of test coverage
   - Tests various JWT claim scenarios and edge cases
   - Validates claim extraction logic and fallback behavior

### Local OIDC Testing:
- Deployed FlightCtl with Keycloak OIDC authentication
- Created and approved enrollment requests via CLI
- **Result:** `APPROVER` correctly shows "demouser" instead of "unknown" ✅

### Unit Test Coverage:
```bash
go test ./internal/auth/... -v
# All tests pass ✅
```

### CODE REVIEW

File: internal/auth/authn/jwtverifer.go [hunk 1]
```diff
@@ -67,8 +68,41 @@ func (j JWTAuth) ValidateToken(ctx context.Context, token string) error {
 }
 
 func (j JWTAuth) GetIdentity(ctx context.Context, token string) (*common.Identity, error) {
```
Comment (DakCrowder) [inline internal/auth/authn/jwtverifer.go:70]: Just a heads up, there are some conflicts added a few weeks ago with this method/file.  There is a new JWTIdentity type that strores the parsed token and ti has a GetClaim method, I'll leave you to sort it out but hopefully fits in nicely with this work.

File: internal/auth/common/common.go [hunk 2]
```diff
@@ -64,3 +64,108 @@ func ExtractBearerToken(r *http.Request) (string, error) {
...
+// - roles as []interface{} (alternative)
+func ExtractGroupsFromJWTClaims(claims map[string]interface{}) []string {
+	var groups []string
+
+	// Try 'groups' claim first (most common)
+	if groupClaim, exists := claims["groups"]; exists {
+		groups = append(groups, parseJWTGroupClaim(groupClaim)...)
+	}
+
+	// Try 'roles' claim as alternative (Azure AD, etc.)
```
Comment (DakCrowder) [inline internal/auth/common/common.go:116]: Should groups and roles be mashed together?  I feel in most cases they are distinct e.g. I might be a member of the engineering group, and within that group I have an administrator or developer role.
